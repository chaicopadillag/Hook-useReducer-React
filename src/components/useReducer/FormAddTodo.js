import React from 'react';
import PropTypes from 'prop-types';
import useForm from '../../hooks/useForm';

const FormAddTodo = ({ handleAddTodo }) => {
	const [{ tarea }, handleChange, reset] = useForm({ tarea: '' });
	const handleSubmit = (e) => {
		e.preventDefault();
		if (tarea !== '') {
			const newTodo = {
				id: new Date().getTime(),
				tarea: tarea,
				estado: false,
			};
			handleAddTodo(newTodo);
			reset();
		}
	};

	return (
		<form method="POST" className="row g-3" onSubmit={handleSubmit}>
			<div className="col-8">
				<input type="text" name="tarea" className="form-control" placeholder="Tarea" value={tarea} onChange={handleChange} />
			</div>
			<div className="col-4 d-grid">
				<button className="btn btn-outline-primary" type="submit">
					Agregar tarea
				</button>
			</div>
		</form>
	);
};

FormAddTodo.propTypes = {
	handleAddTodo: PropTypes.func.isRequired,
};

export default FormAddTodo;
