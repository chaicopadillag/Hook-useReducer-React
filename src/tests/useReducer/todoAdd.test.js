import { shallow } from 'enzyme';
import FormAddTodo from '../../components/useReducer/FormAddTodo';

describe('Prueba de Todo Add', () => {
	const handleAddTodo = jest.fn();
	const wrapper = shallow(<FormAddTodo handleAddTodo={handleAddTodo} />);

	test('debe de mostrar correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('NO debe de llamar handleAddTodo', () => {
		const formSubmit = wrapper.find('form').prop('onSubmit');
		formSubmit({ preventDefault() {} });
		expect(handleAddTodo).toHaveBeenCalledTimes(0);
	});

	test('debe de llamar la funcion handledAddTodo', () => {
		const value = 'Aprender React';
		wrapper.find('input').simulate('change', { target: { value, name: 'tarea' } });
		wrapper.find('form').simulate('submit', { preventDefault() {} });
		expect(handleAddTodo).toHaveBeenCalledTimes(1);
		expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
		expect(handleAddTodo).toHaveBeenCalledWith({
			id: expect.any(Number),
			tarea: value,
			estado: false,
		});
		expect(wrapper.find('input').prop('value')).toBe('');
	});
});
