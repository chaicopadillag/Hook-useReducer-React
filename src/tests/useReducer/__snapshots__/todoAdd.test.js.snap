// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de Todo Add debe de mostrar correctamente 1`] = `
<form
  className="row g-3"
  method="POST"
  onSubmit={[Function]}
>
  <div
    className="col-8"
  >
    <input
      className="form-control"
      name="tarea"
      onChange={[Function]}
      placeholder="Tarea"
      type="text"
      value=""
    />
  </div>
  <div
    className="col-4 d-grid"
  >
    <button
      className="btn btn-outline-primary"
      type="submit"
    >
      Agregar tarea
    </button>
  </div>
</form>
`;
